URL: http://bugs.debian.org/652940

diff -ur mtpfs-0.9.orig/mtpfs.c mtpfs-0.9/mtpfs.c
--- mtpfs-0.9.orig/mtpfs.c	2012-01-03 06:55:55.150753788 -0800
+++ mtpfs-0.9/mtpfs.c	2012-01-03 06:55:57.949730632 -0800
@@ -460,7 +460,7 @@
                 genfile->genre = getGenre (tag);
                 genfile->date = getYear (tag);
                 genfile->usecount = 0;
-                genfile->parent_id = 0;
+                genfile->parent_id = parent_id;
                 genfile->storage_id = 0;
 
                 /* If there is a songlength tag it will take

